plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSynchronizerModule.java

```diff
@@ -41,8 +46,26 @@ public void configure(Binder binder)
 
         // S3
         jsonCodecBinder(binder).bindJsonCodec(S3NativeTransactionLogSynchronizer.LockFileContents.class);
-        synchronizerBinder.addBinding("s3").to(S3NativeTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
-        synchronizerBinder.addBinding("s3a").to(S3NativeTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
-        synchronizerBinder.addBinding("s3n").to(S3NativeTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
+        binder.bind(S3NativeTransactionLogSynchronizer.class).in(Scopes.SINGLETON);
+        binder.bind(S3ConditionalWriteLogSynchronizer.class).in(Scopes.SINGLETON);
+
+        install(conditionalModule(S3FileSystemConfig.class, S3FileSystemConfig::isSupportsExclusiveCreate,
+                s3SynchronizerModule(S3ConditionalWriteLogSynchronizer.class),
+                s3SynchronizerModule(S3NativeTransactionLogSynchronizer.class)));
+    }
+
+    private static Module s3SynchronizerModule(Class<? extends TransactionLogSynchronizer> synchronizerClass)
+    {
+        return new AbstractModule()
+        {
+            @Override
+            protected void configure()
+            {
+                var synchronizerBinder = newMapBinder(binder(), String.class, TransactionLogSynchronizer.class);
```

Comment:
are we using `var` now?

Reply:
In some cased yeah

Reply:
what are the cases? Do we have some guidelines?

Reply:
I'd phrase that like this: when it improves code readability but doesn't make it harder to reason about. 

In this case we know that new*Binder will return a binder, this is common code and common pattern

Reply:
https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-var